botocore/operation.py

```diff
@@ -77,6 +80,30 @@ def call(self, endpoint, **kwargs):
                           model=self.model,
                           params=kwargs)
         request_dict = self.build_parameters(**kwargs)
+
+        service_name = self.service.service_name
+        service_model = self.session.get_service_model(service_name)
+        credentials = self.session.get_credentials()
+        scoped_config = self.session.get_scoped_config()
+        event_emitter = self.session.get_component('event_emitter')
+        resolver = self.session.get_component('endpoint_resolver')
+        scheme = endpoint.host.split(':')[0]
+        if endpoint.region_name is None:
```

Comment:
This code path seems pretty similar to what it was in clients. If you can consolidate some of the code into a utils function that would be nice. But if it is not easy, do not bother since we will be deprecating operations in the near future.